### PR TITLE
Merge rc to beta

### LINKED ISSUE #17036: NVDA is unresponsive/blocking during the timeout period when the update add-on notifies the connection server.
Recently, a large number of users encountered this problem after upgrading to 2024.3 because the Chinese community mirror server ran out of traffic. I think this is the same for not being able to access the NV Access server.

### Steps to reproduce:

1. Install the NVDA Chinese Community Update Mirror add-on and restart NVDA.
2. Wait for the update add-on to check, during which you can keep pressing the key.

### Actual behavior:

When connecting to the server during a timeout. Unable to operate the computer, NVDA is blocked.

### Expected behavior:

Does not block NVDA

### NVDA logs, crash dumps and other attachments:

```log
DEBUG - utils.schedule.callJobOnThread (14:39:00.423) - ScheduleThread (23312):
Starting thread for job: _checkForUpdatableAddons on GUI thread
DEBUG - gui.addonStoreGui.controls.messageDialogs.UpdatableAddonsDialog._checkForUpdatableAddons (14:39:00.424) - MainThread (20796):
checking for updatable add-ons
DEBUG - addonStore.dataManager._DataManager._getCacheHash (14:39:00.425) - MainThread (20796):
Fetching add-on data from https://nvaccess.mirror.nvdadr.com/addonStore//cacheHash.json
IO - inputCore.InputManager.executeGesture (14:39:00.600) - winInputHook (24348):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (14:39:02.382) - winInputHook (24348):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (14:39:02.675) - winInputHook (24348):
Input: kb(laptop):upArrow
...
IO - inputCore.InputManager.executeGesture (14:39:05.445) - winInputHook (24348):
Input: kb(laptop):leftArrow
IO - inputCore.InputManager.executeGesture (14:39:05.461) - winInputHook (24348):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (14:39:05.574) - winInputHook (24348):
Input: kb(laptop):rightArrow
IO - inputCore.InputManager.executeGesture (14:39:05.889) - winInputHook (24348):
Input: kb(laptop):upArrow
DEBUGWARNING - addonStore.dataManager._DataManager._getCacheHash (14:39:21.509) - MainThread (20796):
Unable to get cache hash: HTTPSConnectionPool(host='nvaccess.mirror.nvdadr.com', port=443): Max retries exceeded with url: /addonStore//cacheHash.json (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0BE54770>, 'Connection to nvaccess.mirror.nvdadr.com timed out. (connect timeout=120)'))
DEBUG - addonStore.dataManager._DataManager._getLatestAddonsDataForVersion (14:39:21.510) - MainThread (20796):
Fetching add-on data from https://nvaccess.mirror.nvdadr.com/addonStore//zh_CN/all/2024.3.0.json
DEBUGWARNING - addonStore.dataManager._DataManager._getLatestAddonsDataForVersion (14:39:42.594) - MainThread (20796):
Unable to fetch addon data: HTTPSConnectionPool(host='nvaccess.mirror.nvdadr.com', port=443): Max retries exceeded with url: /addonStore//zh_CN/all/2024.3.0.json (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0D1D4610>, 'Connection to nvaccess.mirror.nvdadr.com timed out. (connect timeout=120)'))
DEBUG - editableText.EditableText._hasCaretMoved (14:39:43.356) - MainThread (20796):
Caret move detected using bookmarks. Elapsed 0 sec, retries 0
```

#### NVDA installed/portable/running from source:
install/portable

#### NVDA version:
2024.3 and versions with updated add-on notifications

#### Windows version:
Windows 11
Version 22H2 (OS internal version 22621.4037)

#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
yes
#### If NVDA add-ons are disabled, is your problem still occurring?
yes, it should also work for cases where you can't access the NV Access server.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

### PR DESCRIPTION
**Pull request by seanbudd**
Merges in #17055 

Fixes #17036

## Summary by CodeRabbit

- **New Features**
	- Enhanced GUI responsiveness by updating the method for displaying dialogs during addon update checks.

- **Bug Fixes**
	- Resolved an issue causing the application to freeze during automatic addon update checks under poor network conditions, improving stability and user experience.

- **Documentation**
	- Added release notes for version 2024.3.1, detailing recent improvements and bug fixes.

### CODE REVIEW

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -88,6 +88,14 @@ Please refer to [the developer guide](https://www.nvaccess.org/files/nvda/docume
   * These keys are currently synchronized until 2025.1.
 * `NVDAObjects.UIA.InaccurateTextChangeEventEmittingEditableText` is deprecated with no replacement. (#16817, @LeonarddeR)
 
+## 2024.3.1
+
+This is a patch release to fix a bug with the automatic add-on update notification.
+
+### Bug Fixes
+
+When automatically checking for add-on updates, NVDA no longer freezes on poor connections. (#17036)
```
Comment (SaschaCowley) [inline user_docs/en/changes.md:97]: Should this be a list item for consistency?
```suggestion
* When automatically checking for add-on updates, NVDA no longer freezes on poor connections. (#17036)
```

**Review by SaschaCowley — CHANGES_REQUESTED** [review]

File: user_docs/en/changes.md (hunk 1 above)
Comment (seanbudd) [reply user_docs/en/changes.md:97]: Good point - I'll have to commit this to RC directly

**Comment by SaschaCowley** [comment]
Can you please update the version in `buildVersion.py`?

**Comment by seanbudd** [comment]
> Can you please update the version in `buildVersion.py`?

It's there on rc: https://github.com/nvaccess/nvda/commit/6ec1c74376130af2c47955843b252d80d1748c9a, however we don't want to update this on beta.

**Comment by SaschaCowley** [comment]
This solution does not work for me: I follow your test steps and NVDA still freezes while checking for updates. I have double-checked and I am running the version with the check on the daemon thread.

**Review by SaschaCowley — APPROVED** [review]

**Pull request merged**